flink-connector-aws/flink-connector-aws-kinesis-streams/src/test/java/org/apache/flink/connector/kinesis/table/util/KinesisProducerOptionsMapperTest.java

```diff
@@ -76,4 +77,19 @@ void testProducerEndpointAndPortExtraction() {
 
         Assertions.assertThat(actualMappedProperties).isEqualTo(expectedOptions);
     }
+
+    @Test
+    void testProducerOptionsMapperDoesNotModifyOptionsInstance() {
+        Map<String, String> deprecatedOptions = new HashMap<>();
+        deprecatedOptions.put("sink.producer.kinesis-endpoint", "some-end-point.kinesis");
+        deprecatedOptions.put("sink.producer.kinesis-port", "1234");
+
+        Map<String, String> deprecatedOptionsImmutable =
+                Collections.unmodifiableMap(deprecatedOptions);
+        Assertions.assertThatNoException()
+                .isThrownBy(
+                        () ->
+                                new KinesisProducerOptionsMapper(deprecatedOptionsImmutable)
+                                        .mapDeprecatedClientOptions());
+    }
```

Comment:
Maybe add similar test for other constructor case?

Reply:
Good point, Thanks